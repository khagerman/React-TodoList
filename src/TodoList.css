.TodoList {
  text-align: center;
}
li {
  list-style: none;
}
