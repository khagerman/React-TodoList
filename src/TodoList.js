import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";
import { v4 as uuid } from "uuid";

const TodoList = () => {
  const [todos, setTodo] = useState([]);

  const renderTodos = () => {
    return (
      <div>
        <ul>
          {todos.map((todo) => (
            <Todo
              key={todo.id}
              id={todo.id}
              todo={todo.todo}
              deleteTodo={deleteTodo}
            />
          ))}
        </ul>
      </div>
    );
  };

  const addTodo = (todo) => {
    let newTodo = { ...todo, id: uuid() };
    setTodo((todos) => [...todos, newTodo]);
  };
  const deleteTodo = (id) => {
    setTodo(todos.filter((t) => t.id !== id));
  };
  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      {renderTodos()}
    </div>
  );
};

export default TodoList;
