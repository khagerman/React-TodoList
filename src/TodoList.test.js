import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new item", function () {
  const { getByLabelText, queryByText } = render(<TodoList />);

  // no items yet
  expect(queryByText("Buy waffles")).not.toBeInTheDocument();

  const todoInput = getByLabelText("New To Do:");

  const submitBtn = queryByText("Add a new To Do!");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "Buy waffles" } });

  fireEvent.click(submitBtn);

  // item exists!
  expect(queryByText("Buy waffles")).toBeInTheDocument();
});
