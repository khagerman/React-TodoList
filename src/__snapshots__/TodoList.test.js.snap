// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form>
      <label
        for="todo"
      >
        New To Do:
      </label>
      <input
        id="todo"
        name="todo"
        value=""
      />
      <button>
        Add a new To Do!
      </button>
    </form>
    <div>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
